### ISSUE TITLE
clone is not entirely replaced by add

### ISSUE THREAD

**Issue by timholy** [JuliaLang/Pkg.jl#629]
If you use `Pkg.clone` you get a "legacy CI script reasons" warning, but it's not so simple.
I had to do this:
```
Pkg.add(Pkg.REPLMode.parse_package("https://github.com/timholy/HeaderREPLs.jl"; add_or_develop=true));
```
If you pass a URL to `Pkg.add` directly you end up triggering
```julia
julia> Pkg.API.check_package_name("https://github.com/timholy/HeaderREPLs.jl")
ERROR: https://github.com/timholy/HeaderREPLs.jl is not a valid packagename
Stacktrace:
 [1] pkgerror(::String) at /home/tim/src/julia-1.0/usr/share/julia/stdlib/v1.0/Pkg/src/Types.jl:120
 [2] check_package_name(::String) at /home/tim/src/julia-1.0/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:22
```

**Comment by KristofferC** [comment]
What you have do to is

```
using Pkg
Pkg.add(PackageSpec(url = "https://github.com/timholy/HeaderREPLs.jl"))
```

There is some help about it

```
help?> Pkg.add
  Pkg.add(pkg::Union{String, Vector{String})
  Pkg.add(pkg::Union{PackageSpec, Vector{PackageSpec}})

  Add a package to the current project. This package will be available using the import and using keywords in the Julia REPL and if the current project is a package, also inside that
  package.

  Examples
  ≡≡≡≡≡≡≡≡≡≡

  Pkg.add("Example") # Add a package from registry
  Pkg.add(PackageSpec(name="Example", version="0.3")) # Specify version
  Pkg.add(PackageSpec(url="https://github.com/JuliaLang/Example.jl", rev="master")) # From url
  Pkg.add(PackageSpec(url="/remote/mycompany/juliapackages/OurPackage"))` # From path (has to be a gitrepo)

  See also PackageSpec.
```

but indeed, the deprecation warning should be improved.

**Comment by 00vareladavid** [comment]
Is there a reason to not use the REPL given your use case?

**Comment by Affie** [comment]
Is there an equivalent function for `Pkg.clone(url::String, name::String)`
My use case is bitbucket only allows lower case names in the url. Therefore, I specify the name explicitly.
eg. `Pkg.clone("..../mypackage.jl.git", "MyPackage")`
Currently Pkg.add uses `mypackage` for the package name instead of `MyPackage`.

**Comment by xgdgsc** [comment]
This add syntax or using REPL is too complicated sometimes. For example, when you collaborate on a centralized jupyterhub and you want to add some package, you don' t want to open terminal and ssh into the server.

**Comment by BioTurboNick** [comment]
I just ran into this issue. Every inconsistency for a common usage creates headaches for new users.

If `]add url` works, so should Pkg.add(url).

**Comment by StefanKarpinski** [comment]
Thanks for the input, but that's not how the relationship between the REPL and the API works. The API is simpler, more basic, more stable and more composable. The REPL is more convenient, has shortcuts and generally more DWIM.

**Comment by StefanKarpinski** [comment]
If you're interested in implementing and maintaining that, you're welcome to join it the fray, but I don't think anyone who works on the package manager currently is inclined to do that.

**Comment by BioTurboNick** [comment]
@StefanKarpinski - Fair enough, I'll throw together a pull request.

@00vareladavid - I'm probably coming at it from a different perspective than you. I'm new to the language, don't know much about API design (clearly). I'm building code that I intend to publish and my interest is seeing that the rough spots I've run into get smoothed over so those who try to use my code and get introduced to Julia don't get hung up in the same spots I did.

My experience with Pkg has been that I found an interesting package, but it wasn't registered and I couldn't figure out how to use Pkg to add it, so I moved on. I may have even seen things that showed the REPL add comment, but not knowing what the Pkg REPL was or how to access it, assumed it would be the same as the function. I didn't really know about the Pkg REPL mode until last week when I looked into making my own package. Only today did I learn from Stefan that the Pkg REPL can also be used in code via `pkg""`

**TL;DR:** IMO it's just painful enough to be a barrier, and ways that are less painful require an extra level of Julia knowledge to discover. And I'll throw something together.

**Comment by 00vareladavid** [comment]
resolved by #1000
